mobius-android/src/main/java/com/spotify/mobius/android/EffectLiveData.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * -\-\-
+ * Mobius
+ * --
+ * Copyright (c) 2017-2018 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+package com.spotify.mobius.android;
+
+import android.arch.lifecycle.LifecycleOwner;
+import android.arch.lifecycle.Observer;
+
+/**
+ * An interface for an object that acts like an Android LiveData, except it will only emit each
+ * object it receives once and only once.<br>
+ * This is meant to be used as a means of sending single-handle Events to the view
+ *
+ * @param <T> The type of object to store
+ */
+public interface EffectLiveData<T> {
```

Comment:
This interface lacks the `removeObserver(Observer<? super T>)` method from `LiveData`. It seems like that would be useful.

Reply:
Or maybe it should be removed, according to the "when in doubt, leave it out" principle (see talk by Josh Bloch or https://dl.acm.org/doi/10.1145/1176617.1176622).

Reply:
The reason I haven't included the `removeObserver` call is that we wrap the given observers into other observers, so it wouldn't be a direct pass-through call to the MutableLiveData, but we would have to somehow keep a map. Keeping a map then means we'll have to worry about the lifecycle of objects (though possibly WeakReference would be enough) - and I didn't want to add that complication into this - which is why I excluded the `removeObserver` call